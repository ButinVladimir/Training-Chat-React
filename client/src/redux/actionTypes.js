export const CHANGE_LOGIN = 'CHANGE_LOGIN';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const CHANGE_SPEECH = 'CHANGE_SPEECH';
export const SAY = 'SAY';
export const ADD_TO = 'ADD_TO';
export const REMOVE_TO = 'REMOVE_TO';
export const CONNECT = 'CONNECT';
export const DISCONNECT = 'DISCONNECT';
export const SHOW_ERROR = 'SHOW_ERROR';
export const HIDE_ERROR = 'HIDE_ERROR';
export const USER_APPEARS = 'USER_APPEARS';
export const USER_DISAPPEARS = 'USER_DISAPPEARS';
export const GET_MESSAGE = 'GET_MESSAGE';
export const UPDATE_USER_LIST = 'UPDATE_USER_LIST';
