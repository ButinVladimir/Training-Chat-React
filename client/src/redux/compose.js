export default function compose(...args) {
  return (state, action) => {
    let resultState = state;
    args.forEach((reducer) => { resultState = reducer(resultState, action); });

    return resultState;
  };
}
