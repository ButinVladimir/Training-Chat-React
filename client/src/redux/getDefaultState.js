export default () => ({
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
  connected: false,
  errorPanel: {
    showError: false,
    error: '',
  },
});
