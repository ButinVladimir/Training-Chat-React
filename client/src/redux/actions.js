import * as actionTypes from './actionTypes';

export const changeLogin = loginValue => ({
  type: actionTypes.CHANGE_LOGIN,
  loginValue,
});

export const login = () => ({
  type: actionTypes.LOGIN,
});

export const logout = () => ({
  type: actionTypes.LOGOUT,
});

export const changeSpeech = speechValue => ({
  type: actionTypes.CHANGE_SPEECH,
  speechValue,
});

export const say = (id, date) => ({
  type: actionTypes.SAY,
  id,
  date,
});

export const addTo = recipient => ({
  type: actionTypes.ADD_TO,
  recipient,
});

export const removeTo = recipient => ({
  type: actionTypes.REMOVE_TO,
  recipient,
});
