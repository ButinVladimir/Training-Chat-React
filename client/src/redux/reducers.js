import compose from './compose';
import loginPanel from './reducers/loginPanel';
import speechPanel from './reducers/speechPanel';

const reducers = compose(loginPanel, speechPanel);

export default (state = {
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
}, action) => reducers(state, action);
