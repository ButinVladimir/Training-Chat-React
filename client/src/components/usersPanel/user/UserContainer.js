import { connect } from 'react-redux';
import User from './User';
import { addTo } from '../../../redux/actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch, ownProps) => ({
  onClickUser: () => dispatch(addTo(ownProps.user)),
});

const UserContainer = connect(mapStateToProps, mapDispatchToProps)(User);

export default UserContainer;
