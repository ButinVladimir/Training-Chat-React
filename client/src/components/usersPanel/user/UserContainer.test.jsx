import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import UserContainer from './UserContainer';
import { addTo } from '../../../redux/actions';

describe('UserContainer', () => {
  const testUser = 'user3';
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
  });

  it('renders without crashing', () => {
    const store = mockStore({});
    const component = mount(
      <Provider store={store}>
        <UserContainer user={testUser} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });

  it('handles click', () => {
    const store = mockStore({});
    const component = mount(
      <Provider store={store}>
        <UserContainer user={testUser} />
      </Provider>,
    );

    component.find('button').simulate('click');

    const actions = store.getActions();
    const expectedAction = [addTo(testUser)];
    expect(actions).toEqual(expectedAction);
  });
});
