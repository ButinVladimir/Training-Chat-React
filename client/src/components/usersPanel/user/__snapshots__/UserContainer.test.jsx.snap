// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContainer renders without crashing 1`] = `
<li
  className="user-line"
  key="user3"
>
  <button
    className="user-select"
    onClick={[Function]}
    type="button"
  >
    user3
  </button>
</li>
`;
